app/models/ability.rb

```diff
@@ -0,0 +1,39 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    if user.admin?
+      can :manage, :all
+    else
+      can :manage, Post, id: Post.where(user_id: user.id).map(&:id)
+      can :manage, Comment, id: Comment.where(user_id: user.id).map(&:id)
+      can :read, :all
+    end
```

Comment:
- [ ] I registered a user1 and logged in successfully, then I tried creating a post but I couldn't. I registered user2 and the same issue occurred. I can't create a post. Kindly fix this by making sure that a user has the ability to create a post. Here is a good [article](https://blog.francium.tech/controlling-authorization-in-rails-with-cancancan-gem-1c2dfeecb26c) on how to add an authorization rule.

Reply:
I fixed the bug, thank you